draft-ietf-quic-transport.md

```diff
@@ -3900,42 +3901,82 @@ determines that the quoted packet has actually been lost.
 
 ## Datagram Packetization Layer PMTU Discovery
 
-Section 6.3 of {{!DPLPMTUD}} provides considerations for implementing Datagram
-Packetization Layer PMTUD (DPLPMTUD) with QUIC.
+When implementing the algorithm in Section 5 of {{!DPLPMTUD}}, the initial value
+of BASE_PMTU SHOULD be consistent with the minimum QUIC packet size. The
+MIN_PLPMTU is the same as the BASE_PMTU.
+
+QUIC endpoints implementing DPLPMTUD maintain a maximum packet size (DPLPMTUD
+MPS) for each combination of local and remote IP addresses.
 
-When implementing the algorithm in Section 5 of {{!DPLPMTUD}}, the initial
-value of BASE_PMTU SHOULD be consistent with the minimum QUIC packet size (1232
-bytes for IPv6 and 1252 bytes for IPv4).
+If a QUIC endpoint determines that the PLPMTU between any pair of local and
+remote IP addresses has fallen below the size needed to support the minimum QUIC
+packet size (BASE_PLPMTU), it MUST immediately cease sending QUIC packets,
+except for DPLPMTUD probe packets, on the affected path. An endpoint MAY
+terminate the connection if an alternative path cannot be found.
 
-PING and PADDING frames can be used to generate PMTU probe packets. These frames
-might not be retransmitted if a probe packet containing them is lost.  However,
-these frames do consume congestion window, which could delay the transmission of
-subsequent application data.
 
-A PING frame can be included in a PMTU probe to ensure that a valid probe is
-acknowledged.
+### DPLPMTUD and Initial Connectivity
 
-The considerations for processing ICMP messages in the previous section also
+From the perspective of DPLPMTUD, QUIC transport is an acknowledged
+packetization layer (PL). A sender can therefore enter the DPLPMTUD BASE state
+when the QUIC connection handshake has been completed and the endpoint has
+established a 1-RTT key.
+
+
+### Sending QUIC DPLPMTUD Probe Packets
+
+DPLPMTU probe packets are ack-eliciting packets.  Probe packets that use the
+PADDING frame therefore implement "Probing using padding data", as defined in
+Section 4.1 of {{!DPLPMTUD}}.  Endpoints could limit the content of probe
+packets to PING and PADDING frames as packets that are larger than the current
```

Comment:
The coalescing part sort of reveals a flaw in the construction though.  It shows that the probe really uses datagrams.  At this level, I think that assuming the hack (coalescing with junk Handshake) is not involved, is OK.  Happy to take suggestions though, because this is definitely awkward.

Reply:
Yes, this was starting to leak badly. I think there's some cleaning up that needs to be done in terms of bringing consistency overall, see meta comment above.